### ISSUE TITLE
Data rendering is not displaying data as expected 

### ISSUE THREAD

**Issue by richiesgr** [paiqo/Databricks-VSCode#124]
Hi 
I'm trying to use with VSC version : 1.74.3 and last available extension version : 1.3.1
When I try to render a table I get this attached a white table showing only [2 items] for each row all other cell are empty.  
Thanks for help
I join a picture to show you 
![Screen Shot 2023-01-15 at 18 35 12](https://user-images.githubusercontent.com/163108/212553667-bab5c18e-23ba-4535-bab4-5444a9851dfe.png)

**Comment by gbrueckl** [comment]
hi, thanks for reporting this, though it technically has nothing to do with this extension but with the used renderer.

it seems like the dataframe you display contained a single column with an array in it and it seems like complex objects (array, struct, map) are not displayed properly.

the output of your last display() command looks unexpected as it should only give you a single column with a string value (which does not match the table displayed)

long story short - complex objects may not be displayed properly according to my experience but this is related to the renderer and not to the Databricks Kernel which just returns a JSON that is then rendered by the renderer

**Comment by richiesgr** [comment]
Hi, thanks for the help
Do you know what kind of object cannot be rendered with the current extension ?

**Comment by gbrueckl** [comment]
I strongly assume its all complex objects (array, struct, map) as most renderers are only designed to display scalar values in a tabular format
would it help to show the JSON definition of those complex columns instead?

**Comment by richiesgr** [comment]
it could help a lot if you know how can integrate a good json viewer(with collapsing etc..) with the extension at this moment I only have the vscode-data-preview available so how can add more option to the "change the presention" list ? 
Thanks

**Comment by gbrueckl** [comment]
notebook renderers can be developed independent of the Databricks VSCode extension
they are just responsible for rendering whatever output the notebook cell delivers
in this case, we deliver a mime-type `application/json` and renderers registered for this mime-type can be used to visualize it

however, there are not too much renderers out there ..

**Comment by gbrueckl** [comment]
these are the only useful renderers I found:
https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter-renderers (probably already installed)
https://marketplace.visualstudio.com/items?itemName=RandomFractalsInc.vscode-data-table (is installed together with this extension)

**Comment by richiesgr** [comment]
Thanks you very much
